src/ds3_connection.c

```diff
@@ -0,0 +1,86 @@
+
+/*
+ * ******************************************************************************
+ *   Copyright 2014-2016 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+#include <string.h>
+#include "ds3.h"
+
+ds3_connection_pool* ds3_connection_pool_init(void) {
+    ds3_connection_pool* pool = g_new0(ds3_connection_pool, 1);
+    pool->next_in = 0;
+    pool->next_out = 0;
+    g_mutex_init(&pool->mutex);
+    g_cond_init(&pool->available_connections);
+    return pool;
+}
+
+void ds3_connection_pool_clear(ds3_connection_pool* pool) {
```

Comment:
This probably should be protected with the mutex to make sure that there isn't something trying to access the pool when attempting to clear it